plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/transformers/api/IResourcePropertySource.java

```diff
@@ -0,0 +1,39 @@
+package org.hotswap.agent.plugin.spring.transformers.api;
+
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.EncodedResource;
+import org.springframework.core.io.support.PropertiesLoaderUtils;
+import org.springframework.core.io.support.ResourcePropertySource;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Properties;
+
+public interface IResourcePropertySource {
```

Comment:
I suggest to simplify this interface as the following code:

1. rename interface name to 'ReloadableResourcePropertySource'
2. remove 'ResourcePropertySource resourcePropertySource();'
3. re-implement reloadPropertySource()

```java
public interface ReloadableResourcePropertySource {

    EncodedResource encodedResource();

    Resource resource();

    default void reloadPropertySource() throws IOException {
        if (!(this instanceof ResourcePropertySource)) {
            return;
        }
        
        if (encodedResource() == null && resource() == null) {
            return;
        }
        
        Properties properties = null;
        if (encodedResource() != null) {
            properties = PropertiesLoaderUtils.loadProperties(encodedResource());
        } else if (resource() != null) {
            try {
                properties = PropertiesLoaderUtils.loadProperties(resource());
            } catch (Exception e) {
                // ignore
            }
        } 
        
        if (properties != null) {
            ResourcePropertySource resourcePropertySource = (ResourcePropertySource) this; 
            Map<String, Object> source = resourcePropertySource.getSource();
            source.clear();
            source.putAll((Map) properties);
        }
    }
}
```

Reply:
done